src/features/search/KennisartikelDetail.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <article>
+    <utrecht-heading model-value :level="level" class="heading">
+      {{ title }}
+    </utrecht-heading>
+    <nav>
+      <ul>
+        <li
+          v-for="({ isActive, label, id }, i) in mappedSections"
+          :key="id + 'nav'"
+          :class="{ 'is-active': isActive }"
+        >
+          <span v-if="isActive">{{ label }}</span>
+          <a v-else :href="`#${id}`" @click.prevent="currentSectionIndex = i">{{
+            label
+          }}</a>
+        </li>
+      </ul>
+    </nav>
+    <section
+      v-for="{ id, html, isActive, label } in mappedSections"
+      :key="id + 'text'"
+      :class="{ 'is-active': isActive }"
+      :id="id"
+    >
+      <utrecht-heading model-value :level="level + 1">{{
+        label
+      }}</utrecht-heading>
+      <div v-html="html"></div>
+    </section>
+  </article>
+</template>
+<script setup lang="ts">
+import { cleanHtml, unEscapeHtml } from "@/helpers/html";
+import { UtrechtHeading } from "@utrecht/web-component-library-vue";
+import { nanoid } from "nanoid";
+import { computed, ref, watch } from "vue";
+
+const language = "nl";
+const knownSections = {
+  specifiekeTekst: "Inleiding",
+  procedureBeschrijving: "Aanvraag",
+  conditions: "Voorwaarden",
+  bewijs: "Bewijs",
+  kostenEnBetaalmethoden: "Kosten",
+  uitersteTermijn: "Termijn",
+  bezwaarEnBeroep: "Bezwaar",
+  vereisten: "Vereisten",
+  notice: "Bijzonderheden",
+  wtdBijGeenReactie: "Geen reactie",
+  contact: "Contact",
+  deskMemo: "KCC",
+} as const;
+
+const uniqueId = nanoid();
+
+const props = defineProps<{
+  jsonObject: any;
+  title: string;
+  level: 2 | 3 | 4;
+}>();
+
+const currentSectionIndex = ref(0);
```

Comment:
This works fine, I think.

However it did make me think about the following: we're currently setting this variable on two different places (in the template map and in the script map), using the `.map(..., i)` index. Could this break if the template or script ever get out of sync?

We have an `uniqueId` variable coupled with the sections, could we use that instead?

Reply:
Very good point. I ended up defining a setActive function in the script, let me know what you think.

Reply:
Ah cool solution, didn't think of this. Works better I think, yes 👍🏼